# Wrangle

In [1]:
import pandas as pd

**Data**

In [2]:
migration = pd.read_excel("https://query.data.world/s/al6wfboprlcexlbsk5a44el7ukdqhr?dws=00000")
migration.head()

Country STATE OR PROVINCE  Year  Inbound  Outbound
0  United States           ALABAMA  2022     0.57      0.43
1  United States           ALABAMA  2021     0.59      0.41
2  United States           ALABAMA  2020     0.61      0.39
3  United States           ALABAMA  2019     0.57      0.43
4  United States           ALABAMA  2018     0.56      0.44

In [3]:
migration.describe()

Year     Inbound    Outbound
count   610.000000  610.000000  610.000000
mean   2017.500000    0.477689    0.510869
std       2.874639    0.100194    0.102179
min    2013.000000    0.000000    0.000000
25%    2015.000000    0.420000    0.450000
50%    2017.500000    0.490000    0.510000
75%    2020.000000    0.540000    0.570000
max    2022.000000    0.750000    1.000000

In [4]:
migration.shape

(610, 5)

In [5]:
migration.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 610 entries, 0 to 609
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Country            610 non-null    object 
 1   STATE OR PROVINCE  610 non-null    object 
 2   Year               610 non-null    int64  
 3   Inbound            610 non-null    float64
 4   Outbound           610 non-null    float64
dtypes: float64(2), int64(1), object(2)
memory usage: 24.0+ KB


In [6]:
# make columns lower case
migration.columns = migration.columns.str.lower().str.replace(" ", "_")
migration.columns

Index(['country', 'state_or_province', 'year', 'inbound', 'outbound'], dtype='object')

In [7]:
# check for null values
migration.isna().sum()

country              0
state_or_province    0
year                 0
inbound              0
outbound             0
dtype: int64

In [8]:
# make year categorical
migration.year = migration.year.astype("category")

In [9]:
migration.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 610 entries, 0 to 609
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   country            610 non-null    object  
 1   state_or_province  610 non-null    object  
 2   year               610 non-null    category
 3   inbound            610 non-null    float64 
 4   outbound           610 non-null    float64 
dtypes: category(1), float64(2), object(2)
memory usage: 20.2+ KB


In [11]:
# save data
migration.to_csv("Household_goods_migration_patterns.csv")

**Take aways**

- 610 rows and 5 columns
- 0 null counts
- values contained in in/out bound are total shipment percetange